### ISSUE TITLE
"close_mosaic" as a custom setting or a custom flag, not a default feature

### ISSUE THREAD

**Issue by karolbadowski** [ultralytics/ultralytics#21008]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello.

In all our trainigns, the disabling of the mosaic augmentation in the end is a source of huge drops of quality (se veral percentage points).

To think of it, it seems like a vary custom and problem-specific / handcrafted optimisation that might be beneficial for one problem (like COCO valset) and not for others. Enabling and disabling of any other augmentations in the end might be beneficial or not beneficial depending on the valset and on the highest frequency of objects sizes/blurryness/rotations/any other features.

Also the moment when this switch-off happens seems handcrafted, as a "magic number" to happen 10 epochs before the end of training.

I have talked with Giovanni and he proposed to create this feature request.

Kindly pelase implement a setting in configuration setting file (optionally also a training flag, but the setting will be the least invasive for starts), where the default behavior could be set / switched on or off.

"close mosaic" could be defined by a number saying how many epochs from the end, this operation happens... and user could set whether it is 10 or 0 or any other number. In case of 0 - that would be for projects where this custom augmentations modification is not to be activated at all (like for us - it is harmful and drops the mAP50-95 by several percentage points).

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @karolbadowski, thank you for your thoughtful feature request and for sharing your experiences with mosaic augmentation in training! 🧩 We appreciate your detailed feedback and suggestions for configurability.

This is an automated response to let you know an Ultralytics engineer will review your request and provide further assistance soon.

In the meantime, we recommend checking out the [Docs](https://docs.ultralytics.com/) for more details on current augmentation options, configuration, and [Python](https://docs.ultralytics.com/usage/python/) or [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

If you have relevant logs, configuration files, or example scripts that illustrate your use case, please consider sharing them here to help us understand your workflow and requirements.

Join our community for discussion and support:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for community knowledge sharing

## Upgrade

To make sure you’re testing with the latest features and fixes, please upgrade to the latest `ultralytics` package (and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date, verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for contributing to Ultralytics! 🚀

**Comment by Y-T-G** [comment]
There's a `close_mosaic` argument which you can set to 0 to disable it:

https://docs.ultralytics.com/usage/cfg/#train-settings
